### PR TITLE
Add release process

### PR DESCRIPTION
**Pull request by LilianBoulard**
A while ago, I wrote this little document, which describes the release process of dirty_cat.

I though it might be useful to add it for future contributors :)

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
@jovan-stojanovic : can you review this. The question that you should be asking yourself is: is this clear enough for you to feel confident in doing a release?

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+To release a new version of dirty_cat,
+here are the main steps and appropriate resources:
+
+1. Update `dirty_cat/CHANGES.rst`. It should be updated at each PR,
+   but double-checking before the release is good practice.
+2. Update `dirty_cat/dirty_cat/VERSION.txt` with the new version number
+3. Update `dirty_cat/setup.py`
+4. You have to commit the changes with a new tag: 
+   the version you're going to push (e.g. `0.1`)
```
Comment (jovan-stojanovic) [inline RELEASE_PROCESS.md:23]: Maybe adding the explicit commands here :
git commit -m "prepare for v1.0.0 release"
git tag v1.0.0
git push

**Review by jovan-stojanovic — CHANGES_REQUESTED** [review]

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (jovan-stojanovic) [reply RELEASE_PROCESS.md:23]: Otherwise, everything is good.

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   `twine upload dist/* --repository-url https://test.pypi.org/legacy/`
+9. Install the new release from the test package index on a dedicated environment
+10. Finally, if that works fine, you can push to the official package index with
+    `twine upload dist/*`
+11. It is also good practice at this point to create a new environment
+    and try installing and using the library.
+12. To finish the procedure, create a new release on the GitHub repository.
+
+### Resources
+
```
Comment (jovan-stojanovic) [inline RELEASE_PROCESS.md:39]: Maybe also add this to resources, may be useful:

[Managing releases in a repository](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)

**Review by jovan-stojanovic — APPROVED** [review]

**Comment by LilianBoulard** [comment]
Thanks for the review, merging !

**Pull request merged**